### PR TITLE
Lighthouse scores part 2 #247

### PR DESCRIPTION
**Pull request by amol-anand**
apologies for the wrong branch name.. 

## Issue:

Fix #247  

## Test URLs
  
- Before: https://main--merative--hlxsites.hlx.page/
- After: https://issue-129-link-titles--merative--hlxsites.hlx.page/
- After: https://issue-129-link-titles--merative--hlxsites.hlx.page/company
- After: https://issue-129-link-titles--merative--hlxsites.hlx.page/documents/case-study/protrial-case-study

### CODE REVIEW

File: scripts/lib-franklin.js [hunk 1]
```diff
@@ -144,6 +144,11 @@ export function decorateIcons(element = document) {
       } else {
         span.innerHTML = iconHTML;
       }
+      const a = span.closest('a');
+      if (a && !a.title) {
+        const iconName = span.classList[1];
+        a.title = iconName.substring(5) || `${a.href}`;
```
Comment (arumsey) [inline scripts/lib-franklin.js:150]: Will  `a.href` do the same thing as ``${a.href}``?

File: scripts/lib-franklin.js [hunk 2]
```diff
@@ -144,6 +144,11 @@ export function decorateIcons(element = document) {
       } else {
         span.innerHTML = iconHTML;
       }
+      const a = span.closest('a');
+      if (a && !a.title) {
+        const iconName = span.classList[1];
```
Comment (arumsey) [inline scripts/lib-franklin.js:149]: You're trusting the class you want will always be second?

File: scripts/lib-franklin.js (hunk 1 above)
Comment (venturewise) [reply scripts/lib-franklin.js:150]: Yes I believe it will.

Comment (arumsey) [inline scripts/lib-franklin.js:150]: I know `5` is meant to represent `icon-` but this has maintenace issues since someone 6 months from now may now know what `5` means. Could you just extract what you want from `span.className` using a RegEx?

File: blocks/header/header.js [hunk 3]
```diff
@@ -285,7 +285,9 @@ export default async function decorate(block) {
   });
 
   // hamburger for mobile
-  const hamburger = createTag('a', { class: 'nav-hamburger', role: 'button', tabindex: '0' });
+  const hamburger = createTag('a', {
+    class: 'nav-hamburger', role: 'button', tabindex: '0', 'aria-label': 'nav-mobile',
```
Comment (venturewise) [inline blocks/header/header.js:289]: Why did you change the order of these, is this significant? (real question, I'm still learning)

File: scripts/lib-franklin.js (hunk 1 above)
Comment (arumsey) [reply scripts/lib-franklin.js:150]: If you need to add a comment to explain what the code is doing that is a sign that the code is probably not very good.

**Comment by arumsey** [comment]
Buttons (with or without icons) are getting both `title` and `aria-label` aplied. Is this the desired behaviour?

image

image

**Comment by arumsey** [comment]
Icon-only links are just getting an `aria-label` applied but the label is the fully qualified HREF. Would it be better if the label was just the `href` attribute?

```
<a href="/contact" aria-label="https---issue-129-link-titles--merative2--hlxsites-hlx-page-contact"><span class="icon icon-email"><svg width="26" height="23.86" viewBox="0 0 26 25" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_615_359)">
    <path d="M3.25 6.14941H22.75V18.8272C22.75 19.025 22.6644 19.2147 22.512 19.3546C22.3597 19.4944 22.153 19.573 21.9375 19.573H4.0625C3.84701 19.573 3.64035 19.4944 3.48798 19.3546C3.3356 19.2147 3.25 19.025 3.25 18.8272V6.14941Z" stroke="black" stroke-width="1.73438" stroke-linecap="round" stroke-linejoin="round"></path>
    <path d="M22.75 6.14941L13 14.3527L3.25 6.14941" stroke="black" stroke-width="1.73438" stroke-linecap="round" stroke-linejoin="round"></path>
    </g>
    <defs>
    <clipPath id="clip0_615_359">
    <rect width="26" height="23.8641" fill="white" transform="translate(0 0.929199)"></rect>
    </clipPath>
    </defs>
    </svg>    
</span></a>
```

File: scripts/scripts.js [hunk 4]
```diff
@@ -483,6 +483,19 @@ function decorateMarketo(main) {
   });
 }
 
+export function addTitleToLinks(el) {
+  el.querySelectorAll('a').forEach((a) => {
+    const ariaLabel = a.getAttribute('aria-label');
```
[the remark this answers is not in the document]
Comment (venturewise) [reply scripts/scripts.js:488]: Google uses title in place of aria-label if there is not one.

**Comment by venturewise** [comment]
> Icon-only links are just getting an aria-label applied but the label is the fully qualified HREF. Would it be better if the label was just the href attribute?

To be honest, I think both are not good. The screen reader is going to attempt to read these. I think I'd rather have "icon". Here's an example of how it should be, in a perfect world: 
image

**Comment by arumsey** [comment]
> > Icon-only links are just getting an aria-label applied but the label is the fully qualified HREF. Would it be better if the label was just the href attribute?
> 
> To be honest, I think both are not good. The screen reader is going to attempt to read these. I think I'd rather have "icon". Here's an example of how it should be, in a perfect world: image

This is why I'm struggling with these PRs. I understand we're trying to keep the LHS happy but by applying arbitrary values that may not actually be useful for accessibility I feel like we're just cheating why the rules were put in place for. If we are going to add aria-labels they should be meaningful and not applied automatically. Authors should be involved to ensure useful labels are added that will actually assist a user.

File: scripts/scripts.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (arumsey) [reply scripts/scripts.js:488]: I would prefer to have `decorateButtons` not set the `title` attribute but instead use `aria-label` when the link does not already contain `textContent`.

[the remark this answers is not in the document]
Comment (venturewise) [reply scripts/scripts.js:488]: After thinking on it, I think re-using the icon name for the parent link's aria-label may be the best option here. 
For linked text, we are covered (since it is text).

**Comment by amol-anand** [comment]
> > > Icon-only links are just getting an aria-label applied but the label is the fully qualified HREF. Would it be better if the label was just the href attribute?
> > 
> > 
> > To be honest, I think both are not good. The screen reader is going to attempt to read these. I think I'd rather have "icon". Here's an example of how it should be, in a perfect world: image
> 
> This is why I'm struggling with these PRs. I understand we're trying to keep the LHS happy but by applying arbitrary values that may not actually be useful for accessibility I feel like we're just cheating why the rules were put in place for. If we are going to add aria-labels they should be meaningful and not applied automatically. Authors should be involved to ensure useful labels are added that will actually assist a user.

Agree. we are only adding something IF the aria-label / title doesn't exist. If it already exists, great. no issue there.

File: scripts/scripts.js [hunk 5]
```diff
@@ -376,15 +376,14 @@ export async function createCard(row, style) {
...
   // Create card div
-  const card = document.createElement('div');
+  const card = createTag('div');
   if (styles.length) {
     styles.forEach((style) => card.classList.add(style));
   }
   // Get category
-  const category = document.createElement('div');
+  const category = createTag('div', { class: 'document-category' });
   category.classList.add('document-category');
```
Comment (venturewise) [inline scripts/scripts.js:387]: You already added the class above in line 386, right?

**Review by venturewise — APPROVED** [review]

File: scripts/scripts.js (hunk 5 above)
Comment (arumsey) [reply scripts/scripts.js]: Good catch. I removed line 386.
